Sources/Account/Account.swift

```diff
@@ -46,12 +46,28 @@ final class Account {
         return knowableBlockCount
     }
 
+    @available(*, deprecated, message: "Use cachedBalance(for:TokenId)")
     var cachedBalance: Balance {
+        cachedBalance(for: .MOB)
+    }
+
+    var cachedTxOutTokenIds: Set<TokenId> {
+        Set(allTxOutTrackers
+            .map { $0.knownTxOut.tokenId })
+    }
+
+    func cachedBalance(for tokenId: TokenId) -> Balance {
         let blockCount = knowableBlockCount
         let txOutValues = allTxOutTrackers
             .filter { $0.receivedAndUnspent(asOfBlockCount: blockCount) }
+            .filter { $0.knownTxOut.tokenId == tokenId }
```

Comment:
This can be done in one pass with && on the two criteria instead of two passes..

Whether or not it matters depends on how many items there might be, and how often cachedBalance might be referenced...

I like the way the chained .filter calls look in the code, just want to call out that it can be a little bit more efficient (if it matters)